File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.java
import java.util.*;

public class Main{
	public static void main(String[]args) {
		Scanner scanner = new Scanner(System.in);
		int array[] = new int [9];
		int sum = 0;
		for(int i=0;i<9;i++) {
			int n = scanner.nextInt();
			sum+=n;
			array[i] = n;
		}
		Arrays.sort(array);
		//9개 합에서 2개를 뺀다
exit:	for(int i=0;i<array.length;i++) {
			int total = sum;
			for(int j=i+1;j<array.length;j++) {
				total = total - array[i] - array[j];
				if(total == 100) {					
					array[i] = 0;
					array[j] = 0;
					break exit;
				}
				total = sum;
			}
		}
		for(int i=0;i<array.length;i++) {
			if(array[i]!=0) {
				System.out.println(array[i]);
			}
		}
	}
}

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);
	int sum = 0, arr[9];
	for (int i = 0; i < 9; i++) {
		cin >> arr[i];
		sum += arr[i];
	}
	sort(arr, arr + 9);//미리 정렬
	for (int j = 0; j < 8; j++) {
		int total = sum;
		for (int k = j + 1; k < 9; k++) {
			
			total = total - arr[j] - arr[k];
			if (total == 100) {
				for (int z = 0; z < 9; z++) {
					if (z != j && z != k)
						cout << arr[z] << "\n";
				}
				return 0;
			}
			total = sum;
		}
	}

}